frontend/src/composables/useElementId.ts

```diff
@@ -0,0 +1,33 @@
+import { Ref, readonly, ref } from "vue"
+
+function createUseElementId() {
+  /**
+   * Used for keeping track of existing IDs in the unlikely event that
+   * we generate a duplicate one.
+   */
+  const currentIds = new Set<string>()
+
+  function getIdentifier() {
```

Comment:
I'm quite unsure how good it is that the id now starts with a number and includes a `.` as both is (while technically supported) not recommended as there are a couple of pitfalls that one could fall into with them: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id